nbfetch/handlers.py

```diff
@@ -2,29 +2,42 @@
 from tornado.escape import url_escape, url_unescape
 import traceback
 from urllib.parse import urljoin
-from notebook.base.handlers import IPythonHandler
+
 import threading
 import json
 import os
+from pathlib import Path
+import io
 from queue import Queue, Empty
-import jinja2
 from hs_restclient import HydroShare, HydroShareAuthBasic, HydroShareAuthOAuth2
 from .pull import GitPuller, HSPuller
 from .version import __version__
-from notebook.utils import url_path_join
 import pickle
+from jupyter_server.base.handlers import JupyterHandler
+from jupyter_server.extension.handler import (
+    ExtensionHandlerMixin,
+    ExtensionHandlerJinjaMixin,
+)
```

Comment:
Looks like most of the changes to this file are formatting changes.  From what I can see there are a couple of new imports for new base classes that needed very little modification from the old class and some extra documentation around authentication.

Reply:
Yeah, sorry that I made it kind of a pain to review. I should have turned off my code formatter for PR readability. Thanks for the review, @sblack-usu! :tada: